src/modules/flow/http-server/http-server.c

```diff
@@ -157,30 +121,18 @@ boolean_response_cb(void *data, struct sol_http_request *request)
         }
     }
 
-    if (send_json) {
-        r = snprintf(response_str, sizeof(response_str), "{\"%s\": %s}", mdata->path,
-            mdata->value.b == true ? "true" : "false");
-    } else {
-        r = snprintf(response_str, sizeof(response_str), "%s", mdata->value.b == true ? "true" : "false");
-    }
-
+    r = type->response_cb(mdata, &response.content, send_json);
     SOL_INT_CHECK_GOTO(r, < 0, end);
 
     r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
```

Comment:
I'd move this inside `response_cb()` as well, change to send the whole `&response` instead of just `&response.content`.  Then you don't need to use `send_json` in here. And you allow responses to add extra headers/parameters as they wish.